src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.impl.response.ShoppingCartResponseMapper;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+    private final MovieSessionService movieSessionService;
+    private final ShoppingCartResponseMapper responseMapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  UserService userService,
+                                  MovieSessionService movieSessionService,
+                                  ShoppingCartResponseMapper responseMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+        this.responseMapper = responseMapper;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long id) {
+        return responseMapper.toDto(shoppingCartService.getByUser(userService.get(id)));
+    }
+
+    @PutMapping("/movie-sessions")
+    @ResponseStatus(value = HttpStatus.OK)
+    public void addMovieSession(@RequestParam Long userId,
```

Comment:
Maybe you can use "addSession" name for this method to keep consistency with service layer method